From 94631bfd9b813d9d2f1af0ea744e7121d72d25c5 Mon Sep 17 00:00:00 2001
From: Alberto Milone <alberto.milone@canonical.com>
Date: Fri, 10 Jul 2015 16:52:29 +0200
Subject: [PATCH 1/2] Add support for Linux 4.1

Drop the IRQF_DISABLED flag, as it was removed in Linux 4.1

Signed-off-by: Alberto Milone <alberto.milone@canonical.com>
---
 firegl_public.c | 4 +++-
 kcl_acpi.c      | 4 ++++
 2 files changed, 7 insertions(+), 1 deletion(-)

--- 15.7/common/lib/modules/fglrx/build_mod/firegl_public.c
+++ 15.7b/common/lib/modules/fglrx/build_mod/firegl_public.c
@@ -3495,10 +3495,12 @@ int ATI_API_CALL KCL_InstallInterruptHandler(
         KCL_PUB_InterruptHandlerWrap,
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,22)
         ((useMSI) ? (SA_INTERRUPT) : (SA_SHIRQ)),
-#else
+#elif LINUX_VERSION_CODE < KERNEL_VERSION(4,1,0)
         //when MSI enabled. keep irq disabled when calling the action handler,
         //exclude this IRQ from irq balancing (only on one CPU) 
         ((useMSI) ? (IRQF_DISABLED) : (IRQF_SHARED)),    
+#else
+        ((useMSI) ? (0x0) : (IRQF_SHARED)),
 #endif
         dev_name,
         context);